mlir/lib/Bindings/Python/IRModule.h

```diff
@@ -748,6 +748,31 @@ class PyRegion {
   MlirRegion region;
 };
 
+/// Wrapper around an MlirAsmState.
+class PyAsmState {
+ public:
+  PyAsmState(MlirValue value, bool useLocalScope) {
+    flags = mlirOpPrintingFlagsCreate();
+    // The OpPrintingFlags are not exposed Python side, create locally and
+    // associate lifetime with the state.
+    if (useLocalScope)
+      mlirOpPrintingFlagsUseLocalScope(flags);
+    state = mlirAsmStateCreateForValue(value, flags);
+  }
+  ~PyAsmState() {
+    mlirOpPrintingFlagsDestroy(flags);
+  }
+  // Delete copy constructors.
+  PyAsmState(PyAsmState &other) = delete;
+  PyAsmState(const PyAsmState &other) = delete;
```

Comment:
just for my own edification about "best practices" - is the de facto reason for this so that there's no references hanging around to something e.g., the `MlirAsmState state`?

Reply:
Its to avoid accidentally copying this and then having the flags get destroyed multiple times in the copy's destructor (I was using a wrong accessor and ended up copying these, so avoid the footgun for myself :)).